### DISCUSSION TITLE
톰캣이란 무엇인가?

### DISCUSSION THREAD

**Discussion by alstn113** [2024-woowacourse-study/level-interview#243] [Spring Boot]
.

**Comment by alstn113** [comment]
### Apache 웹 서버와 Tomcat 웹 애플리케이션 서버

#### Apache 웹 서버

- 제작 및 배포: 아파치 소프트웨어 재단에서 제작한 웹 서버로, 리눅스에서는 `httpd`로 배포됨.
- 기능: 클라이언트 요청(예: HTML, CSS, 이미지 파일 등 정적 데이터)을 처리하여 응답.
- 사용 포트: 기본적으로 80번 포트를 사용.
- 특징: HTTP 프로토콜을 기반으로 웹 클라이언트(브라우저)의 요청을 서비스함.

#### Tomcat 웹 애플리케이션 서버(WAS)

- 제작 및 배포: 아파치 소프트웨어 재단에서 제작한 서블릿 컨테이너로, 웹 서버와 웹 컨테이너의 결합.
- 기능: JSP와 서블릿 처리, 서블릿의 수명 주기 관리, 요청 URL을 서블릿 코드로 매핑, HTTP 요청 수신 및 응답, 필터 체인 관리 등을 담당.
- 사용 포트: 기본적으로 8080 포트를 사용.
- 특징: 현재 가장 많이 사용되는 웹 애플리케이션 서버(WAS)로, 동적인 데이터 처리 및 비즈니스 로직 수행에 적합.

#### 웹 서버(Web Server)와 웹 애플리케이션 서버(WAS)의 차이점

- 웹 서버:
    - 역할: 정적 데이터(HTML, 이미지 등) 요청 처리에 특화.
    - 장점: 빠르고 안정적.
    - 예: Apache HTTP Server.

- 웹 애플리케이션 서버(WAS):
    - 역할: 동적 데이터 처리, DB 연결, 데이터 조작 등의 비즈니스 로직 수행에 적합.
    - 장점: 비즈니스 로직 수행에 효율적.
    - 예: Tomcat, WebLogic, Jeus, JBoss.

#### Apache Tomcat (Apache + Tomcat)

- 통합 기능: Tomcat은 편의를 위해 Apache의 일부 기능(예: 웹 서비스 데몬, Httpd)을 포함하고 있어 "Apache Tomcat"이라고 불림.
- 분업 사용: 특정 상황에서는 Apache 서버와 Tomcat 서버를 함께 사용하여 로드 밸런싱, PHP 애플리케이션 지원 등의 요구를 충족시킴.
    - 예: PHP 애플리케이션과 자바 애플리케이션을 함께 사용하거나 Httpd 서버를 로드 밸런싱에 사용.


참고: https://inpa.tistory.com/entry/TOMCAT-⚙%EF%B8%8F-설치-설정-정리

**Comment by 3Juhwan** [comment]
**Apache Tomcat**

자바 기반의 서블릿 컨테이너이자 웹 서버입니다. WAS로써 웹 서버 역할을 하면서 동적인 자바 웹 애플리케이션을 실행할 수 있는 환경을 제공합니다. 

Apache Tomcat을 이해하기 위해 Apache와 Tomcat에 대한 이해가 필요합니다. 

**Apache**

- 아파치 소프트웨어 재단에서 만든 웹서버입니다. 
- 웹서버로써 정적 리소스를 제공합니다. 
- 80 포트를 사용합니다. 

**Tomcat**

- 웹 서버와 웹 컨테이너의 결합한 형태입니다. 
- 현재 가장 일반적이고 많이 사용되는 WAS입니다. 
- 컨테이너, 웹 컨테이너, 서블릿 컨테이너라고도 부릅니다. 
- JSP와 서블릿 처리, 서블릿의 수명 주기 관리, 요청 URL을 서블릿 코드로 매핑, HTTP 요청 수신 및 응답, 필터 체인 관리 등을 처리해준다.
- 8080 포트를 사용합니다. 


![web server & was](https://blog.kakaocdn.net/dn/dzROHQ/btrinP7Hryy/s0OadohZFwNMwbVqVXhGI1/img.png)

**Comment by le2sky** [comment]
![image](https://github.com/user-attachments/assets/3f459456-e9e9-49fb-a7d7-507ea93091bb)
